pkg/asset/agent/gencrypto/authconfig.go

```diff
@@ -303,8 +312,8 @@ func (a *AuthConfig) refreshAuthTokenSecret(k8sclientset kubernetes.Interface, r
 	return nil
 }
 
-// GetAuthTokenFromCluster returns a token string stored as the secret from the cluster.
-func GetAuthTokenFromCluster(ctx context.Context, kubeconfigPath string) (string, error) {
+// GetWatcherAuthTokenFromCluster returns a watcherAuth token string stored as the secret from the cluster.
+func GetWatcherAuthTokenFromCluster(ctx context.Context, kubeconfigPath string) (string, error) {
```

Comment:
I see there is a unit test for the install workflow. I suggest creating unit tests for the addnodes workflow. One of the unit tests should verify if the tokens are generated if they were previously generated and expired.

Reply:
I would like to add the new unit tests via seperate PR